wordpress-mobile/WordPress-Android
# Fix WP Screenshot Automation UI Tests

Fixes #16845 Partially — only the `WPScreenshotTest` and with the code for taking the editor screenshot commented out.

Changes:
- Add missing mock for the wpcom v2 dashboard cards request of site `106707880`.
- Fix navigation to post list.
- Fix navigation to stats.
- Fix navigation to media and back.
- Make navigating back from media more stable.
- Add navigation utilities to `MySitesPage` and use them.
- Create method for going to drafts in `PostsListPage` and use it.
- Refactor code to remove `moveToStats` - Ideally there'll be only methods for screenshots in `WPScreenshotTest`. More granular methods that concern only specific tests add to the cognitive load and make it harder to understand the class.
- Comment out `editBlogPost` and add info explaining why.

### To test:
 - Run the `wPScreenshotTest` from `WPScreenshotTest` 5 times in Android Studio to check that it succeeds consecutively and it's pretty stable.

 - Ideally we run them on an emulator created by `fastlane`. See the notes in the issue for how this can be done:
   > Create a virtual device, or
Follow the guide in `SCREENSHOT_DEVICE_SETUP.md` to create a device like the ones we're usin for screenshots automation (this won't be super easy as the guide is a bit outdated and if you're on an M1 Mac you have to change the generated .ini files manually to set the arm64 architecture or else the emultators won't start.


## Regression Notes
1. Potential unintended areas of impact
   The code I changed has only one usage which is this test.
   I'm not sure how the `fastlane screenshot` command will behave with the editor screenshot missing.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   N/a

3. What automated tests I added (or what prevented me from doing so)
   N/a

PR submission checklist:
